src/components/select/select.component.ts

```diff
@@ -324,9 +324,10 @@ export class SelectComponent<T> implements OnInit, OnChanges, OnDestroy, Control
             event.preventDefault();
         }
 
-        // when the user types and the value is not empty then we should open the dropdown
-        if (event.keyCode !== ESCAPE) {
+        // when the user types and the value is not empty then we should open the dropdown except for non printable keys.
+        if (event.ctrlKey === false && event.shiftKey === false && event.altKey === false && event.metaKey === false && !(event.keyCode === TAB) && !(event.keyCode === ESCAPE)) {

```

Comment:
There is an `event.key` property that is available, for printable characters it returns the letter/number, for not printable keys it returns the name, eg: `control` etc.. If the key length is 1 then we can tell if it is a printable key or not